artichoke-backend/src/extn/core/time/trampoline.rs

```diff
@@ -1,35 +1,151 @@
 //! Glue between mruby FFI and `Time` Rust implementation.
 
-use spinoso_time::MICROS_IN_NANO;
-
-use crate::convert::implicitly_convert_to_int;
-use crate::extn::core::time::Time;
+use crate::convert::{implicitly_convert_to_int, implicitly_convert_to_string};
+use crate::extn::core::symbol::Symbol;
+use crate::extn::core::time::{Offset, Time};
 use crate::extn::prelude::*;
 
-// Constructor
+const MAX_NANOS: i64 = 1_000_000_000 - 1;
+const MILLIS_IN_NANO: i64 = 1_000_000;
+const MICROS_IN_NANO: i64 = 1_000;
+const NANOS_IN_NANO: i64 = 1;
+
+// Generate a subsecond multiplier from the given ruby value.
+//
+// - If not provided, the defaults to Micros.
+// - Otherwise, expects a symbol with :milliseconds, :usec, or :nsec.
+fn subsec_multiplier(interp: &mut Artichoke, subsec_type: Option<Value>) -> Result<i64, Error> {
+    subsec_type.map_or(Ok(MICROS_IN_NANO), |mut value| {
+        let subsec_symbol = unsafe { Symbol::unbox_from_value(&mut value, interp)? }.bytes(interp);
+        match subsec_symbol {
+            b"milliseconds" => Ok(MILLIS_IN_NANO),
+            b"usec" => Ok(MICROS_IN_NANO),
+            b"nsec" => Ok(NANOS_IN_NANO),
+            _ => Err(ArgumentError::with_message("unexpected unit. expects :milliseconds, :usec, :nsec").into()),
+        }
+    })
+}
```

Comment:
I just discovered `Time.at(1, 0,75)` is a thing:

```
irb(main):001:0> Time.at(1, 0.75)
=> 1970-01-01 01:00:01.00000075 +0100
```

Which is 0.75 micros (the default if nil), or 750 nanos. Of course I had the naive "this must just be an integer" kind of thoughts, but of course it's the `to_int` dance too! :dancers: 

```
irb(main):001:0> class A; def to_int; 1000; end; end
=> :to_int
irb(main):002:0> Time.at(0, A.new).tv_nsec
=> 1000000
```

How do you feel about:

```rust
struct Subsec(i64);

impl TryConvertMut<(Option<Value>, Option<Value>, Subsec> for Artichoke {
    type Error = Error;

    fn try_convert_mut(&mut self, subsec_and_type: (Option<Value>, Option<Value>)) -> Result<Subsec, Self::Error> {
        todo!();
    }
}
```

:point_up: This would take both the subsecond param, and the optional subsecond type.

Could also use the `SubsecMultiplier` in conjunction with this too of course, but it feels like this logic needs to be pulled out a little.

Reply:
ooof fun times with implicit conversions. This looks like a good approach to me. ooc, did you find this when playing with the specs?

Reply:
Yep! The specs lead me to it, so they are actually quite helpful! I apparently missed the 5th line of implementation in the ruby docs, but I must admit even they are light on this: https://ruby-doc.org/core-2.6.3/Time.html#method-c-at